src/main/java/org/mule/extensions/jms/internal/connection/provider/GenericConnectionProvider.java

```diff
@@ -93,4 +122,59 @@ public JmsSupport create102bSupport() {
       }
     };
   }
+
+  protected void configureSSLContextIfNeeded() {
+    if (!trustStorePassword.isEmpty() && !trustStoreName.isEmpty()) {
+      try {
+        final SSLContext context = SSLContext.getInstance(PROTOCOL);
+        context.init(new KeyManager[0],
+                     getCustomTrustStoreWithDefaultCerts(getTruststoreFile(trustStoreName), trustStorePassword),
+                     new SecureRandom());
+        SSLContext.setDefault(context);
+      } catch (Exception e) {
+        throw new JmsExtensionException("Failed to set custom TrustStore", e);
+      }
+    }
+  }
+
+  private TrustManager[] getCustomTrustStoreWithDefaultCerts(java.util.Optional<File> truststoreFile, String trustStorePassword)
+      throws Exception {
+    final KeyStore keyStore = getKeyStoreWithCustomCerts(truststoreFile, trustStorePassword);
+
+    final TrustManagerFactory jdkDefaultTrustManagerFactory =
+        TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+    jdkDefaultTrustManagerFactory.init((KeyStore) null);
+    List<TrustManager> trustManagers = Arrays.asList(jdkDefaultTrustManagerFactory.getTrustManagers());
+
+    List<X509Certificate> certificates = trustManagers.stream().filter(X509TrustManager.class::isInstance)
+        .map(X509TrustManager.class::cast).map(trustManager -> Arrays.asList(trustManager.getAcceptedIssuers()))
+        .flatMap(Collection::stream).collect(Collectors.toList());
+
+    for (X509Certificate certificate : certificates) {
+      keyStore.setCertificateEntry(String.valueOf(certificate.getSerialNumber()), certificate);
+    }
+    final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+    trustManagerFactory.init(keyStore);
+
+    return trustManagerFactory.getTrustManagers();
+  }
+
+  private KeyStore getKeyStoreWithCustomCerts(java.util.Optional<File> truststoreFile, String trustStorePassword)
+      throws CertificateException, IOException, NoSuchAlgorithmException, KeyStoreException {
+    final KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+    keyStore.load(null, null);
+    try (FileInputStream kos = new FileInputStream(truststoreFile.get())) {
```

Comment:
The `truststoreFile` parameter is an Optional, you should check if it isPresent() before accessing it.

Reply:
I added a catch for this case.